### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by TheFakeOG**
Finished foundation for testing
Scratched mocha/chai, now using karma/jasmine

**Last 7 commits overwrites all previous commits, so there is no need to view the previous commits**

### CODE REVIEW

**Review by KevinMidboe — COMMENTED** [review]
Changes to mqtt.connect in home.ts and mqttClient.ts so it now uses the input for username, host adr and port from user input. Everything else added is for testing proposes.

**Review by KevinMidboe — APPROVED** [review]
Changes to mqtt.connect in home.ts and mqttClient.ts so it now uses the input for username, host adr and port from user input. Everything else added is for testing proposes.

File: GATEWAY/README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@
 ### Install cross-platform tools for CLI-commands (Needed to build and test the solution)
 (This is needed because the development-team works on both IOS and Windows, and needed a copy-command that works on both platforms)
 
-`$ npm install -g cd-cli`
+`$ npm install -g cp-cli`
```
Comment (nornes) [inline GATEWAY/README.md:13]: This is no longer used and can be removed entirely

File: GATEWAY/src/app/app.component.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { Component } from '@angular/core';
+import { Component, Injectable, Inject } from '@angular/core';
```
Comment (nornes) [inline GATEWAY/src/app/app.component.ts:1]: Why are these needed?

**Review by nornes — COMMENTED** [review]
Just a few comments
